server/controller/semester.controller.js

```diff
@@ -34,36 +40,49 @@ semesterController.post(
   passport.authenticate('jwt', { session: false }),
   validateCreateSemester,
   async (req, res) => {
-    const errorsAfterValidation = validationResult(req);
-    if (!errorsAfterValidation.isEmpty()) {
-      res.status(400).json({
-        code: 400,
-        errors: errorsAfterValidation.mapped(),
-      });
-    } else {
+    validationHandler(req, res, async () => {
       try {
-        const { term, year, courses } = req.body;
-        const semester = await Semester.findOne({ term, year, courses });
+        const semester = await Semester.findOne(req.body);
 
         if (!semester) {
-          const newSemester = await createSemester(req.body);
-          res.status(200).json(newSemester);
-        } else {
-          generateServerErrorCode(
-            res,
-            403,
-            'create semester error',
-            SEMESTER_EXISTS_ALREADY,
-            'semester_id'
-          );
+          const newSemester = await semesterController.createSemester(req.body);
+          return res.status(200).json(newSemester);
         }
+        return generateServerErrorCode(
+          res,
+          403,
+          'create semester error',
+          SEMESTER_EXISTS_ALREADY,
+          'semester_id'
+        );
       } catch (e) {
-        generateServerErrorCode(res, 500, e, SOME_THING_WENT_WRONG);
+        return generateServerErrorCode(res, 500, e, SOME_THING_WENT_WRONG);
       }
-    }
+    });
   }
 );
 
+const getSemesterWithOption = (res, option = {}) => {
```

Comment:
Same comment from course controller: Possibly rename getSemesterWithOption() to getPopulatedSemester() to exemplify that you are populating based on the options argument.

Reply:
Additionally, move this constant definition to the top of the file with other constants.